### PR TITLE
[client] recover wallet on start if there is a mnemonic_file

### PR DESCRIPTION
**Pull request by yuan-xy**
## Motivation

When user start client with "-n, --mnemonic-file <mnemonic-file>" parameter, we should try recover wallet from the mnemonic-file.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)
Start Libra client with a mnemonic-file been set:
```
cargo run --bin client -- --host ac.testnet.libra.org -n <some mnemonic-file>
```
if the mnemonic-file is valid, You will seen output like this:
```
Wallet recovered and the first 3 child accounts were derived
#0 address 7af57a0c206fbcc846532f75f373b5d1db9333308dbc4673c5befbca5db60e2f
#1 address f1f48f56c4deea75f4393e832edef247547eb76e1cd498c27cc972073ec4dbde
#2 address 8b1d500b1e9b43ad0ba244ff109b558eb503b486cf760f5782a71dd7aee09d5a
Connected to validator at: localhost:49516, latest version = 11826, timestamp = 2020-01-02 15:17:43.787651 UTC
```

cargo xtest

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: libra-swarm/src/client.rs [hunk 1]
```diff
@@ -155,7 +155,7 @@ impl InProcessTestClient {
                     .unwrap(),
                 false,
                 /* faucet server */ None,
-                Some(mnemonic_file_path.to_string()),
+                &Some(mnemonic_file_path.to_string()),
```
Comment (davidiw) [inline libra-swarm/src/client.rs]: What's the motivation of passing by reference?

File: client/src/client_proxy.rs [hunk 2]
```diff
@@ -802,7 +802,7 @@ impl ClientProxy {
         Ok(())
     }
 
-    /// Recover wallet accounts from file and return vec<(account_address, index)>.
+    /// Recover wallet accounts from space_delim_strings and return vec<(account_address, index)>.
```
Comment (davidiw) [inline client/src/client_proxy.rs]: space_delim_strings should be written out "space delimited strings" or even better yet giving an example of the formatting.

**Comment by yuan-xy** [comment]
I've made two improvements:
1. copy "mnemonic_file" and pass it to client_proxy by value instead of by reference.
2. rename "space_delim_strings" to "command 'recover <file>'" in comment.

File: client/src/main.rs [hunk 3]
```diff
@@ -97,6 +97,20 @@ fn main() -> std::io::Result<()> {
         "Connected to validator at: {}:{}, {}",
         args.host, args.port, ledger_info_str
     );
+    if let Some(file) = args.mnemonic_file {
+        match client_proxy.recover_wallet_accounts_by_file(&file) {
```
Comment (davidiw) [inline client/src/main.rs]: I should have said this earlier, apologies, but why aren't we doing this within the new function of the client proxy. It seems weird that the ClientProxy takes this file but doesn't perform the recovery itself.

Maybe we could update ClientProxy::get_libra_wallet to call recover_wallet_accounts_by_file?

Other than this, this PR looks good :)

**Comment by yuan-xy** [comment]
Yes, it seems weird. So I change the `recover_wallet_accounts_by_file` to `recover_accounts_in_wallet`and reuse the wallet object already instantiated by `get_libra_wallet`.

`recover_accounts_in_wallet` should be an instance method which updates accounts in `ClientProxy` and `get_libra_wallet` is a static method. So we can't do it in `get_libra_wallet`, we already do too much things in the constructor of ClientProxy which is not a good design.

**Comment by yuan-xy** [comment]
And sync_wallet with network should be done after network connection testing, so I put `recover_accounts_in_wallet` after the `test_validator_connection` method call.

**Comment by davidiw** [comment]
Fair enough, the design is rough and needs some love, but this at least helps make it re-usable. Thanks!

@bors-libra r+

**Pull request closed without merging**